src/plugins/auto/src/auto_schedule.cpp

```diff
@@ -139,10 +142,50 @@ void AutoSchedule::init() {
             // if have CPU Device,  enable m_compile_context[CPU]
             if (cpu_iter != m_context->m_device_priorities.end()) {
                 m_compile_context[CPU].m_is_enabled = true;
-                m_compile_context[CPU].m_device_info = *cpu_iter;
-                m_compile_context[CPU].m_device_info.config[ov::hint::performance_mode.name()] = ov::hint::PerformanceMode::LATENCY;
-                m_compile_context[CPU].m_worker_name = "CPU_HELP";
-                LOG_INFO_TAG("will load CPU for accelerator");
+                if (is_actual_gpu) {
+                    // user does not set the compiling threads
+                    // limit the threads num for compiling
+                    auto device = m_compile_context[ACTUALDEVICE].m_device_info.device_name;
+                    auto& device_config = m_compile_context[ACTUALDEVICE].m_device_info.config;
+                    int max_threads = 0;
+                    try {
+                        max_threads = m_context->m_ov_core->get_property(device, ov::compilation_num_threads);
+                    } catch (const ov::Exception&) {
+                        LOG_DEBUG_TAG("cannot get MAX_NUM_THREADS from GPU");
+                    }
+                    if (max_threads == static_cast<int>(std::thread::hardware_concurrency())) {
+                        int thread_num = max_threads / 2;
+                        m_compile_context[ACTUALDEVICE].m_device_info.config.insert(
+                            ov::compilation_num_threads(thread_num));
+                        LOG_DEBUG_TAG("gpu streams number for compiling: %d", thread_num);
+                    } else {
+                        // user set the compiling threads num
+                        // use the user's val anyway
+                        LOG_DEBUG_TAG("user defined compiling threads: %d", max_threads);
+                    }
+                    std::string cache_dir = device_config.count(ov::cache_dir.name())
+                                                ? device_config[ov::cache_dir.name()].as<std::string>()
+                                                : m_context->m_ov_core->get_property("", ov::cache_dir);
+
+                    if (!m_context->m_is_set_startup_fallback && !cache_dir.empty()) {
```

Comment:
This logic applies not only to GPUs but to all HW accelerator, for example, AUTO:NPU,CPU

Reply:
Updated.